Formula/mesa.rb

```diff
@@ -0,0 +1,368 @@
+      special(int k, int x, int y)
+      {
+        switch (k) {
+        case GLUT_KEY_UP:
+          view_rotx += 5.0;
+          break;
+        case GLUT_KEY_DOWN:
+          view_rotx -= 5.0;
+          break;
+        case GLUT_KEY_LEFT:
+          view_roty += 5.0;
+          break;
+        case GLUT_KEY_RIGHT:
+          view_roty -= 5.0;
+          break;
+        default:
+          return;
+        }
+        glutPostRedisplay();
+      }
+
+      /* new window size or exposure */
+      static void
+      reshape(int width, int height)
+      {
+        GLfloat h = (GLfloat) height / (GLfloat) width;
+
+        glViewport(0, 0, (GLint) width, (GLint) height);
+        glMatrixMode(GL_PROJECTION);
+        glLoadIdentity();
+        glFrustum(-1.0, 1.0, -h, h, 5.0, 60.0);
+        glMatrixMode(GL_MODELVIEW);
+        glLoadIdentity();
+        glTranslatef(0.0, 0.0, -40.0);
+      }
+
+      static void
+      init(void)
+      {
+        static GLfloat pos[4] =
+        {5.0, 5.0, 10.0, 0.0};
+        static GLfloat red[4] =
+        {0.8, 0.1, 0.0, 1.0};
+        static GLfloat green[4] =
+        {0.0, 0.8, 0.2, 1.0};
+        static GLfloat blue[4] =
+        {0.2, 0.2, 1.0, 1.0};
+
+        glLightfv(GL_LIGHT0, GL_POSITION, pos);
+        glEnable(GL_CULL_FACE);
+        glEnable(GL_LIGHTING);
+        glEnable(GL_LIGHT0);
+        glEnable(GL_DEPTH_TEST);
+
+        /* make the gears */
+        gear1 = glGenLists(1);
+        glNewList(gear1, GL_COMPILE);
+        glMaterialfv(GL_FRONT, GL_AMBIENT_AND_DIFFUSE, red);
+        gear(1.0, 4.0, 1.0, 20, 0.7);
+        glEndList();
+
+        gear2 = glGenLists(1);
+        glNewList(gear2, GL_COMPILE);
+        glMaterialfv(GL_FRONT, GL_AMBIENT_AND_DIFFUSE, green);
+        gear(0.5, 2.0, 2.0, 10, 0.7);
+        glEndList();
+
+        gear3 = glGenLists(1);
+        glNewList(gear3, GL_COMPILE);
+        glMaterialfv(GL_FRONT, GL_AMBIENT_AND_DIFFUSE, blue);
+        gear(1.3, 2.0, 0.5, 10, 0.7);
+        glEndList();
+
+        glEnable(GL_NORMALIZE);
+      }
+
+      void
+      visible(int vis)
+      {
+        if (vis == GLUT_VISIBLE)
+          glutIdleFunc(idle);
+        else
+          glutIdleFunc(NULL);
+      }
+
+      main(int argc, char *argv[])
+      {
+        glutInit(&argc, argv);
+        if (argc > 1) {
+          /* do 'n' frames then exit */
+          limit = atoi(argv[1]) + 1;
+        } else {
+          limit = 0;
+        }
+        glutInitDisplayMode(GLUT_RGB | GLUT_DEPTH | GLUT_DOUBLE);
+
+        glutCreateWindow("Gears");
+        init();
+
+        glutDisplayFunc(draw);
+        glutReshapeFunc(reshape);
+        glutKeyboardFunc(key);
+        glutSpecialFunc(special);
+        glutVisibilityFunc(visible);
+
+        glutMainLoop();
+        return 0;             /* ANSI C requires main to return int. */
+      }
+    EOS
+    system "cc", "#{testpath}/gears.c", "-o gears", "-L#{lib}", "-L#{HOMEBREW_PREFIX}/lib/", "-lglut", "-lgl"
```

Comment:
Use `Formula["freeglut"].opt_lib` instead of `HOMEBREW_PREFIX`

Reply:
I'm not sure how to correctly use double quotes inside a double quotes?